Fix parallel make

marcadella opened the pull request.

fabern commented:
@marcadella could you explain (briefly) what these changes did? 
It looks like you could simplify the Make file by a lot.

marcadella commented:
Firstly I removed Makevars.win because the compilation process is platform agnostic so there is no reason to duplicate the makefile specifically for Windows. Secondly, I removed half of the targets because they were not needed in the sense the the recipe for getting the .o also produces the .mod, so having a target for the .mod files was not needed. Thirdly I added a few targets which were missing and cause parallel make to fail.
